src/main/java/seedu/address/logic/parser/UndoCommandParser.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.UndoCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new UndoCommand object
+ */
+public class UndoCommandParser implements Parser<UndoCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the UndoCommand
+     * and returns an UndoCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public UndoCommand parse(String args) throws ParseException {
+        if (args.isBlank()) {
+            return new UndoCommand();
+        }
+
+        try {
+            Index index = ParserUtil.parseIndex(args);
+            return new UndoCommand(index);
+        } catch (ParseException pe) {
+            throw new ParseException(MESSAGE_INVALID_COMMAND_FORMAT);
```

Comment:
Can add message of the correct usage of the Undo Command to the user as well. Something similar to the `DeleteCommandParser` class, like for example:
`throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UndoCommand.MESSAGE_USAGE), pe);`